qiskit_dynamics/backend/backend_utils.py

```diff
@@ -167,3 +170,67 @@ def _sample_probability_dict(
 def _get_counts_from_samples(samples: list) -> Dict:
     """Count items in list."""
     return dict(zip(*np.unique(samples, return_counts=True)))
+
+
+def _iq_data(
+    options: Options,
+    state: Union[Statevector, DensityMatrix],
+    measurement_subsystems: List[int],
+    seed: int,
+) -> List[List[List[float]]]:
```

Comment:
I think for functions at this level you should make the specific fields of `options` they require explicit arguments. So make `subsystem_dims`, `iq_centers`, and `iq_widths` args of this function.

Reply:
Another question for this function is: do the memory slots play any role here? Can the IQ data be put into different orderings (other than the explicit qubit ordering)?

Reply:
So I ran some circuits on hardware, and yes, the measure order does make a difference. Also, for unused `cl_bits`, we get an array of `0+0j` with the length of shots. Should we have similar behaviour?

Reply:
I was trying to understand the code (here)[https://github.com/rupeshknn/qiskit-dynamics/blob/19482c5d62e918d72a551791164f3ed67bee3ff4/qiskit_dynamics/backend/backend_utils.py#L142-L145]
```
if memory_slot_result in memory_slot_probs:
        memory_slot_probs[memory_slot_result] += prob
else:
        memory_slot_probs[memory_slot_result] = prob
```
the code under `else` is clear, but what are we expecting to do with the `if` code

Reply:
The loop this lives in is translating the probabilities of observing the state in different computational basis states (limited to some subsystems) into probabilities for memory slot outcomes. So, e.g. if the state-outcome string is `'01'`, and the first subsystem is being mapped to memory slot 2, and the second to memory slot 0, (with memory slot 0 being unused), the corresponding memory slot string is then `'100'` (where all of these strings are in reverse subsystem ordering).

The reason for the `if` statement is that if there's a max outcome level, then the mapping of state-outcome to measurement_slot-outcome is not 1 to 1. E.g. if the max outcome level is `1`, the state-outcome string `'02'` will also be mapped to the memory-slot string `'100'`. So the if statement is checking if the string already exists in the dictionary of memory slot outcome probabilities: if it doesn't then put it in, and if it does exist, add the current probability to the existing probability.

This all being said, this block could be reduced to the single line:
```
memory_slot_probs[memory_slot_result] = prob + memory_slot_probs.get(memory_slot_result, 0)
```

Reply:
done